staker-rewards/lib/assets-calculator.js

```diff
@@ -0,0 +1,182 @@
+import { callWithRetry, getPastEvents } from "./contract-helper.js"
+
+export default class AssetsCalculator {
+  constructor(
+    context,
+    interval,
+    tokenStaking,
+    keepRandomBeaconOperator,
+    bondedECDSAKeepFactory,
+    keepBonding,
+    sortitionPoolAddress
+  ) {
+    this.context = context
+    this.interval = interval
+    this.tokenStaking = tokenStaking
+    this.keepRandomBeaconOperator = keepRandomBeaconOperator
+    this.bondedECDSAKeepFactory = bondedECDSAKeepFactory
+    this.keepBonding = keepBonding
+    this.sortitionPoolAddress = sortitionPoolAddress
+  }
+
+  static async initialize(context, interval) {
+    const { contracts } = context
+
+    const bondedECDSAKeepFactory = await contracts.BondedECDSAKeepFactory.deployed()
+    const tbtcSystem = await contracts.TBTCSystem.deployed()
+
+    const sortitionPoolAddress = await callWithRetry(
+      bondedECDSAKeepFactory.methods.getSortitionPool(
+        tbtcSystem.options.address
+      )
+    )
+
+    const assetsCalculator = new AssetsCalculator(
+      context,
+      interval,
+      await contracts.TokenStaking.deployed(),
+      await contracts.KeepRandomBeaconOperator.deployed(),
+      bondedECDSAKeepFactory,
+      await contracts.KeepBonding.deployed(),
+      sortitionPoolAddress
+    )
+
+    await assetsCalculator.fetchBondEvents()
+
+    return assetsCalculator
+  }
+
+  async calculateOperatorAssets(operator) {
+    const keepStaked = await this.calculateKeepStaked(operator)
+    const ethBonded = await this.calculateETHBonded(operator)
+    const ethUnbonded = await this.calculateETHUnbonded(operator)
+    const ethTotal = Math.round((ethBonded + ethUnbonded) * 100) / 100
```

Comment:
Do we need to round it? We should use the precise number for reward calculation, if possible. We can display them rounded in the table though.

Reply:
The same comment applies to all other rounding operations in this file.